Doc/tools/check-warnings.py

```diff
@@ -94,9 +235,10 @@ def fail_if_improved(
 def main() -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "--check-and-annotate",
-        help="Comma-separated list of files to check, "
-        "and annotate those with warnings on GitHub Actions",
+        "--check-and-annotate-changed",
```

Comment:
Perhaps call it `*-diff` so we know to give it two `git diff` type args?

```suggestion
        "--check-and-annotate-diff",
```

Reply:
Sure; I ended up renaming it to the shorter `annotate-diff` since it doesn't actually perform the checks itself, just control whether annotations are emitted between the specified/default refs.